torchio/data/image.py

```diff
@@ -262,16 +264,35 @@ def get_bounds(self):
         return bounds_x, bounds_y, bounds_z
 
     @staticmethod
-    def _parse_path(path: TypePath) -> Path:
+    def _parse_path(
+            path: Union[TypePath, Sequence[TypePath]]
+            ) -> Union[Path, List[str]]:
         if path is None:
             return None
-        try:
-            path = Path(path).expanduser()
-        except TypeError:
-            message = f'Conversion to path not possible for variable: {path}'
-            raise TypeError(message)
-        if not (path.is_file() or path.is_dir()):  # might be a dir with DICOM
-            raise FileNotFoundError(f'File not found: {path}')
+
+        if isinstance(path, (str, Path)):
+            try:
+                path = Path(path).expanduser()
+            except TypeError:
+                message = f'Conversion to path not possible for variable: ' \
+                          f'{path}'
+                raise TypeError(message)
+        else:
+            try:
+                path = [Path(p).expanduser() for p in path]
+            except TypeError:
+                message = f'Conversion to list of paths not possible for ' \
+                          f'variable: {path}'
+                raise TypeError(message)
+
+        if isinstance(path, Path):
```

Comment:
Can the input be a list of dirs?

Reply:
I don't know, it seems that for a single path a dir was acceptable, why not for a list of paths? Even if it corresponds to images with several channels it will work as long as they have the same spatial dimensions.

Reply:
It's just the impression I got by skimming through the code here. I'll get a better sense of it once I've properly gone through it.